src/Apps/Order/Hooks/useStripePaymentBySetupIntentId.tsx

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from "react"
+import { useSystemContext } from "System"
+import { getClientParam } from "Utils/getClientParam"
+import { SetPaymentByStripeIntent } from "Apps/Order/Mutations/SetPaymentByStripeIntent"
+
+/*
+ * Hook to handle Stripe redirect for newly-linked bank account
+ * pulls necessary params from Stripe redirect URL and sets payment by intentId
+ */
+export function useStripePaymentBySetupIntentId(orderId: string) {
+  const { relayEnvironment } = useSystemContext()
+  const [isProcessingRedirect, setIsProcessingRedirect] = useState(true)
+  const [stripeSetupIntentId, setStripeSetupIntentId] = useState<null | string>(
+    null
+  )
+  const [isPaymentSetupSuccessful, setIsPaymentSetupSuccessful] = useState(
+    false
+  )
+
+  useEffect(() => {
+    const setPaymentBySetupIntentId = async (
+      setupIntentId: string,
+      oneTimeUse: boolean
+    ) => {
+      try {
+        if (!relayEnvironment) return
+
+        const error = (
+          await SetPaymentByStripeIntent(relayEnvironment, {
+            id: orderId,
+            oneTimeUse,
+            setupIntentId,
+          })
+        ).commerceSetPaymentByStripeIntent?.orderOrError?.error
+
+        if (error) throw error
+        setIsPaymentSetupSuccessful(true)
+      } catch (error) {
+        setIsPaymentSetupSuccessful(false)
+      }
+    }
+
+    // pull necessary params from Stripe redirect URL
+    const saveAccount = getClientParam("save_account")
```

Comment:
For router consistency (and SSR support) we should always grab this off of the router itself, versus the window:
```tsx
const { match } = useRouter()

match.location... 
````